docs/source/user_guide/gcm_based_inference/answering_causal_questions/computing_counterfactuals.rst

```diff
@@ -66,20 +70,32 @@ As we see, with :math:`X = 2` and :math:`Y \approx 4.03`, :math:`Z` should be ap
 we know the hidden noise for this sample, approximately -2.98. So the counterfactual outcome
 is again :math:`Z \approx 9.07`.
 
+.. _understand-method:
+
 Understanding the method
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
-Counterfactuals are very similar to :doc:`simulate_impact_of_interventions`, with an important
+Counterfactuals in graphical causal models are very similar to :doc:`simulate_impact_of_interventions`, with an important
 difference: when performing interventions, we look into the future, for counterfactuals we look into
-an alternative past. To reflect this in the computation, when performing interventions, we generate
-all noise using our causal models. For counterfactuals, we use the noise from actual observed data.
-
-To expand on our example above, we assume there are other factors that contribute to cholesterol
-levels, e.g. exercising or genetic predisposition. While we *assume* medication helps against high
-LDL levels, it's important to take into account all other factors that could also help against it.
-We want to prove *what* has helped. Hence, it's important to use the noise from the real data,
-not some generated noise from our generative models. Otherwise, I may be able to reduce my
-cholesterol LDL level in the counterfactual world, where I take medication (X := 5), but not because
-I took the medication, but because the *generated noise* of Z also just happened to be low and so
-caused a low value for Z. By taking the *real* noise value of Z (derived from the observed data of
-Z), I can prove that it was the medication that helped.
+an alternative past. To reflect this in the computation, when performing interventions, we first recreate all noise values
```

Comment:
Invertibility leads to point counterfactuals. But we may also have distributional counterfactuals, where we estimate the posterior distribution of noise terms given the observational evidence. We should make the latter explicit here as most readers of Pearl are familiar with the notion of distributional counterfactuals.

Reply:
Good point, I added a note at the bottom. Allowing counterfactual distributions (beyond point-wise) is anyway something high on the todo list.